### PR TITLE
Fixed deprecation warnings

### PR DESCRIPTION
**Pull request by ilyachur**

### CODE REVIEW

File: demos/crossroad_camera_demo/main.cpp [hunk 1]
```diff
@@ -161,23 +161,19 @@ struct PersonDetection : BaseDetection{
...
-    CNNNetwork read() override {
+    CNNNetwork read(const Core& ie) override {
         slog::info << "Loading network files for PersonDetection" << slog::endl;
-        CNNNetReader netReader;
         /** Read network model **/
-        netReader.ReadNetwork(FLAGS_m);
-        /** Extract model name and load it's weights **/
-        std::string binFileName = fileNameNoExt(FLAGS_m) + ".bin";
-        netReader.ReadWeights(binFileName);
+        auto network = ie.ReadNetwork(FLAGS_m);
```
Comment (IRDonch) [inline demos/crossroad_camera_demo/main.cpp]: I think you should specify the path to the weights explicitly here. The documentation says that if the second argument is omitted and a file with the automatically-derived name isn't found, the function will not load the weights at all, which is not what we want. We want it to fail with a file-not-found error.

Comment (ilyachur) [reply demos/crossroad_camera_demo/main.cpp]: @IRDonch if IR contains some weighs but you don't provide **\*.bin** file, the network won't be read and you will get error that some layer cannot set weights.

Comment (IRDonch) [reply demos/crossroad_camera_demo/main.cpp]: I have experimented with this, and the errors you get when you specify the weights path and when you don't are quite different.

When you don't, `ReadNetwork` just crashes.

When you do, it throws an exception:

```
terminate called after throwing an instance of 'InferenceEngine::details::InferenceEngineException'
  what():  filesize for: test.bin - -1<0. Please, check weights file existence.
```

The latter is certainly preferable to the former, as it hints as to what went wrong.

Comment (ilyachur) [reply demos/crossroad_camera_demo/main.cpp]: I created a PR to OpenVINO which add more informative error message if model requires weights, but bin file wasn't found.

Comment (IRDonch) [reply demos/crossroad_camera_demo/main.cpp]: All right then, sounds good to me.

File: demos/mask_rcnn_demo/main.cpp [hunk 2]
```diff
@@ -103,14 +103,10 @@ int main(int argc, char *argv[]) {
         // --------------------Load network (Generated xml/bin files)-------------------------------------------
         slog::info << "Loading network files" << slog::endl;
 
-        InferenceEngine::CNNNetReader networkReader;
         /** Read network model **/
-        networkReader.ReadNetwork(FLAGS_m);
+        auto network = ie.ReadNetwork(FLAGS_m);
 
         /** Extract model name and load weights **/
```
Comment (IRDonch) [inline demos/mask_rcnn_demo/main.cpp]: Orphaned comment.

File: demos/multi_channel/object_detection_demo_yolov3/main.cpp [hunk 3]
```diff
@@ -244,20 +244,20 @@ DisplayParams prepareDisplayParams(size_t count) {
     return params;
 }
 
-std::map<std::string, YoloParams> GetYoloParams(const std::vector<std::string>& outputDataBlobNames, 
-                                                InferenceEngine::CNNNetReader &netReader) {
+std::map<std::string, YoloParams> GetYoloParams(const std::vector<std::string>& outputDataBlobNames,
+                                               InferenceEngine::CNNNetwork &network) {
```
Comment (IRDonch) [inline demos/multi_channel/object_detection_demo_yolov3/main.cpp]: Restore the alignment, please.

File: demos/object_detection_demo_faster_rcnn/main.cpp [hunk 4]
```diff
@@ -112,19 +112,12 @@ int main(int argc, char *argv[]) {
         // -----------------------------------------------------------------------------------------------------
 
         // --------------------------- 2. Read IR Generated by ModelOptimizer (.xml and .bin files) ------------
-        std::string binFileName = fileNameNoExt(FLAGS_m) + ".bin";
         slog::info << "Loading network files:"
             "\n\t" << FLAGS_m <<
-            "\n\t" << binFileName <<
             slog::endl;
```
Comment (IRDonch) [inline demos/object_detection_demo_faster_rcnn/main.cpp]: It's misleading to say "network files" but only list one file.

File: demos/pedestrian_tracker_demo/main.cpp [hunk 5]
```diff
@@ -77,7 +76,7 @@ CreatePedestrianTracker(const std::string& reid_model,
         tracker->set_descriptor_strong(descriptor_strong);
         tracker->set_distance_strong(distance_strong);
     } else {
-        std::cout << "WARNING: Either reid model or reid weights "
+        std::cout << "WARNING: Either reid model "
```
Comment (IRDonch) [inline demos/pedestrian_tracker_demo/main.cpp]: You can't use "either" with only one item.

**Comment by IRDonch** [comment]
Jenkins: test this please

**Comment by IRDonch** [comment]
@Wovchena Please review.

The build failures on CI are likely due to an old OpenVINO version. I'll update it soon.

**Comment by IRDonch** [comment]
Jenkins please retry a build

File: demos/human_pose_estimation_demo/src/human_pose_estimator.cpp [hunk 6]
```diff
@@ -34,10 +34,7 @@ HumanPoseEstimator::HumanPoseEstimator(const std::string& modelPath,
         ie.SetConfig({{InferenceEngine::PluginConfigParams::KEY_PERF_COUNT,
                        InferenceEngine::PluginConfigParams::YES}});
     }
-    netReader.ReadNetwork(modelPath);
-    std::string binFileName = fileNameNoExt(modelPath) + ".bin";
-    netReader.ReadWeights(binFileName);
-    network = netReader.getNetwork();
+    auto network = ie.ReadNetwork(modelPath);
```
Comment (IRDonch) [inline demos/human_pose_estimation_demo/src/human_pose_estimator.cpp]:
```suggestion
    network = ie.ReadNetwork(modelPath);
```

You've shadowed the member variable with the same name.

**Comment by IRDonch** [comment]
Jenkins: test this please

**Pull request merged**